src/bokeh/models/widgets/tables.py

```diff
@@ -858,6 +858,37 @@ def __init__(self, *args, **kwargs) -> None:
     The height of each row in pixels.
     """)
 
+    @staticmethod
+    def to_data_table(data, columns=None, formatters={}, **kwargs) -> DataTable:
```

Comment:
I feel like `DataTable.to_data_table` could be a confusing ("why would I turn a data table into a data table?") Maybe this could be re-framed as a "from" function
```suggestion
    def from(data, columns=None, formatters={}, **kwargs) -> DataTable:
```
or similar. Any thoughts @tcmetzger ?